lib/slim_fast/parser.ex

```diff
@@ -101,7 +101,7 @@ defmodule SlimFast.Parser do
              |> String.split(~r/^(-|==|=)/)
              |> List.last
              |> String.lstrip
-    inline = inline or String.starts_with?(input, "=")
+    inline = inline or String.starts_with?(input, "=") or String.starts_with?(script, "if")
```

Comment:
Does this also match `- ifoobar`? Probably a low risk of false matches, but still – maybe we could bound the match.

There are maybe a few other expressions to consider: unless, case, cond, for comprehensions? If we want to match Ruby Slim behaviour, I guess things like `- Enum.each foos, fn (x) ->` should work as well, but I'm not sure whether or not we support that even with a `=`.

If we want to support all those things (basically any function or macro…) I suspect we'd need to make bigger changes in the library. I haven't looked into how this stuff works.

I think in the case of Ruby Slim, it's "simply" the case that

```
- whatever do |x|
  = x.foo
```

becomes something equivalent to `<% whatever do |x| %> <%= x.foo %><% end %>`, which doesn't treat the method as something that returns a value, but only concatenates whatever `=` lines of code it happens to execute into some output buffer.

Maybe that's not a good match for how this library works, or indeed how Elixir works? Since it's a lot about everything returning values. Again, I haven't looked into these parts of the code, so I don't really know.